llvm/cmake/config-ix.cmake

```
@@ -17,6 +55,150 @@ include(CheckCompilerVersion)
 include(CheckProblematicConfigurations)
 include(HandleLLVMStdlib)
 
+if (ANDROID OR DRAGONFLY OR FREEBSD OR HAIKU OR LINUX OR NETBSD OR OPENBSD OR SOLARIS)
+  set(HAVE_DLFCN_H 1)
+  set(HAVE_ERRNO_H 1)
+  set(HAVE_FCNTL_H 1)
+  set(HAVE_FENV_H 1)
+  set(HAVE_LINK_H 1)
+  set(HAVE_MACH_MACH_H 0)
+  set(HAVE_MALLOC_MALLOC_H 0)
+  set(HAVE_PTHREAD_H 1)
+  set(HAVE_SIGNAL_H 1)
+  set(HAVE_SYS_IOCTL_H 1)
+  set(HAVE_SYS_MMAN_H 1)
+  set(HAVE_SYS_PARAM_H 1)
+  set(HAVE_SYS_RESOURCE_H 1)
+  set(HAVE_SYS_STAT_H 1)
+  set(HAVE_SYS_TIME_H 1)
+  set(HAVE_SYS_TYPES_H 1)
+  set(HAVE_SYSEXITS_H 1)
+  set(HAVE_TERMIOS_H 1)
+  set(HAVE_UNISTD_H 1)
+elseif (AIX)
+  set(HAVE_DLFCN_H 1)
+  set(HAVE_ERRNO_H 1)
+  set(HAVE_FCNTL_H 1)
+  set(HAVE_FENV_H 0)
+  set(HAVE_LINK_H 0)
+  set(HAVE_MACH_MACH_H 0)
+  set(HAVE_MALLOC_MALLOC_H 0)
+  set(HAVE_PTHREAD_H 1)
+  set(HAVE_SIGNAL_H 1)
+  set(HAVE_SYS_IOCTL_H 0)
+  set(HAVE_SYS_MMAN_H 0)
+  set(HAVE_SYS_PARAM_H 1)
+  set(HAVE_SYS_RESOURCE_H 0)
+  set(HAVE_SYS_STAT_H 1)
+  set(HAVE_SYS_TIME_H 0)
+  set(HAVE_SYS_TYPES_H 1)
+  set(HAVE_SYSEXITS_H 0)
+  set(HAVE_TERMIOS_H 1)
+  set(HAVE_UNISTD_H 1)
```

This is based on what I had from my `CMakeCache.txt` on AIX:
```suggestion
  set(HAVE_DLFCN_H 1)
  set(HAVE_ERRNO_H 1)
  set(HAVE_FCNTL_H 1)
  set(HAVE_FENV_H 1)
  set(HAVE_LINK_H 0)
  set(HAVE_MACH_MACH_H 0)
  set(HAVE_MALLOC_MALLOC_H 0)
  set(HAVE_PTHREAD_H 1)
  set(HAVE_SIGNAL_H 1)
  set(HAVE_SYS_IOCTL_H 1)
  set(HAVE_SYS_MMAN_H 1)
  set(HAVE_SYS_PARAM_H 1)
  set(HAVE_SYS_RESOURCE_H 1)
  set(HAVE_SYS_STAT_H 1)
  set(HAVE_SYS_TIME_H 1)
  set(HAVE_SYS_TYPES_H 1)
  set(HAVE_SYSEXITS_H 1)
  set(HAVE_TERMIOS_H 1)
  set(HAVE_UNISTD_H 1)
```

---

Thank you for checking! I suspected as much, but there's a room for improvement of AIX documentation.